sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClientBuilder.java

```diff
@@ -180,6 +182,24 @@ public NettyAsyncHttpClientBuilder configuration(Configuration configuration) {
         return this;
     }
 
+    /**
+     * Disables deep copy of response {@link ByteBuffer} into a heap location that is managed by this client as
+     * opposed to the underlying netty library which may use direct buffer pool.
+     * <br>
+     * <b>
+     * Caution: Disabling this is not recommended as it can lead to data corruption if the downstream consumers
+     * of the response do not handle the byte buffers before netty releases them.
+     * </b>
+     *
+     * @param disableBufferCopy If set to {@code true}, the client built from this builder will not deep-copy
+     * response {@link ByteBuffer ByteBuffers}.
+     * @return The updated {@link NettyAsyncHttpClientBuilder} object.
+     */
+    public NettyAsyncHttpClientBuilder disableBufferCopy(boolean disableBufferCopy) {
```

Comment:
Also This would be a good candidate to add sample in `samples\README.md` showing how user can use this.
Also changelog entry for this change.

Reply:
Since this behavior is not something we want users to eagerly use, I don't want to put this in README. I have the javadoc updated with codesnippets. So, if they really want to use, they can find it's usage in javadocs.